### PR TITLE
CB-13810: Validate the existence of key url provided by user for Azure cmk

### PR DESCRIPTION
**Pull request by juhi-09**
User provides the encryption key url during creation of environment to enable disk encryption with CustomerManagedKey. The existence of key on Azure cloud is checked as part of this PR.
The getKey method in AzureClient either returns the keyBundle if the key exists or returns null.

### CODE REVIEW

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java [hunk 1]
```diff
@@ -46,6 +46,10 @@
 
     private static final Pattern ENCRYPTION_KEY_URL_VAULT_NAME = Pattern.compile("https://([^.]+)\\.vault.*");
 
+    private static final Pattern ENCRYPTION_VAULT_URI = Pattern.compile("(.*)\\/keys\\/.*");
+
+    private static final Pattern ENCRYPTION_KEY_NAME = Pattern.compile("https:.*\\/keys\\/([^.]+)\\/.*");
+
```
Comment (cegganesh84) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java:52]: Suggestion to consider : Is it possible to use single regex like `"(https://([^.]+).vault.[^/]+)/keys/([^/]+)/.*"` where group 1 will be vault uri, 2 will be vault name and 3 will be key name?

Also `\\/` I guess these double escaping is not needed, it should work without those.

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java [hunk 2]
```diff
@@ -1021,4 +1022,17 @@ public void deleteDiskEncryptionSet(String resourceGroup, String diskEncryptionS
         return azureClientCredentials.getAccesToken();
     }
 
+    public boolean checkEncryptionKeyExistenceOnCloud(String vaultUri, String keyName, String keyVaultName, String keyVaultResourceGroupName) {
+        return handleAuthException(() -> {
+            PagedList<KeyItem> keys = azure.vaults()
+                    .getByResourceGroup(keyVaultResourceGroupName, keyVaultName)
+                    .client()
+                    .listKeys(vaultUri);
```
Comment (cegganesh84) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java:1030]: Is my understanding right that the results are paginated and if the key is not present on the first page we will return `false`?

Comment (juhi-09) [reply cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java:1030]: Thank you for pointing this out. Fixing this.

**Review by cegganesh84 — APPROVED** [review]
Just a nit: does not exists -> does not exist

**Pull request merged**